e2e/tests/start_call.spec.ts

```diff
@@ -77,6 +77,30 @@ test.describe('start new call', () => {
         expect(await page.locator('#calls-widget .calls-widget-bottom-bar').screenshot()).toMatchSnapshot('dm-calls-widget-bottom-bar.png');
         await devPage.leaveCall();
     });
+
+    test('slash command from existing thread', async ({page, context}) => {
+        // create a test thread
+        await page.locator('#post_textbox').fill('test thread');
+        await page.locator('[data-testid=SendMessageButton]').click();
+        const post = page.locator('.post-message__text').last();
+        await expect(post).toBeVisible();
+
+        // open RHS
+        await post.click();
+        await expect(page.locator('#rhsContainer')).toBeVisible();
+
+        // send slash command in thread to start a call.
+        await page.locator('#reply_textbox').fill('/call start');
+        await page.locator('#reply_textbox').press('Enter');
+        await expect(page.locator('#calls-widget')).toBeVisible();
+
+        // verify the call post is created in the thread.
+        await expect(page.locator('#rhsContainer').filter({has: page.getByText(`${userState.users[0].username} started a call`)})).toBeVisible();
```

Comment:
I guess it wouldn't hurt, I'll have a look.

Reply:
@cpoile Just to clarify, starting a second call from the same thread is allowed. What's not allowed is having two calls at the same time but that goes beyond the changes here, it was never possible to start with. Were you thinking of testing such case?

Reply:
Added a test for the "starting call twice in same channel" case although not from the call thread since [there's a bug](https://mattermost.atlassian.net/browse/MM-50398) preventing the error from showing.

Reply:
Ah, that's weird, I didn't receive a notification for that ping, sorry for the late reply. I meant the former -- just the second call in the same thread. Ohh, https://mattermost.atlassian.net/browse/MM-50398 is a good find. 

Hmm... But why would there be an error in the RHS is we're starting a call from an existing call thread (when the previous call is already finished)?

Reply:
@cpoile Apologies for the confusion. I am still talking about starting a call with an on-going call, which should error. The other case is allowed as you can have as many subsequent calls as you wish on the same thread since we are essentially creating call posts instead of threads. Do you find that unexpected or is there something I am missing here?

Reply:
Nope, that's clears it up, and now it seems expected at least to me (but maybe it's needs some PM/design input?).

Reply:
My thinking was that if we don't allow this behaviour then the whole changes are much less useful. I think one of the originating goals was to keep a recurring thread for subsequent calls but happy to hear thoughts of course as it's easy enough to change.

@itao Let me know if you have concerns/ideas on this.

Reply:
If I understand correctly, the behaviour here is:
- There can still only be one ongoing call per channel at a time
- This lets someone start a call in an existing thread, regardless if the thread was previously used for another call
- If there is already an ongoing call in the channel, regardless if it's in this or another thread, then an error tells the user to join that call instead
- As a user, this lets me escalate from text to voice without losing context of the discussion so far when viewing thread instead of the channel (ie. from threads inbox, RHS, etc)

Is that correct? If so, that makes sense to me and sound great. Could that error for existing call be more helpful by asking if the user would like to join the call instead, as opposed to just flat out error?

Reply:
@itao

> - There can still only be one ongoing call per channel at a time
> - This lets someone start a call in an existing thread, regardless if the thread was previously used for another call
> - As a user, this lets me escalate from text to voice without losing context of the discussion so far when viewing thread instead of the channel (ie. from threads inbox, RHS, etc)

Yes to all points.

> - If there is already an ongoing call in the channel, regardless if it's in this or another thread, then an error tells the user to join that call instead

This is where it gets more tricky as we are still relying on a suboptimal errors system native of the webapp. Essentially when a slash command fails for any reason we return an error and then the webapp side handles it by showing it on the bottom of the center channel view:

![image](https://user-images.githubusercontent.com/1832946/217850872-fedf2244-d16c-44cc-81e8-f369b2655ae6.png)

Personally I am not in love with this but historically it was the easiest way to get some sort of feedback to the user so we never tried to do anything better. On top of this there's a [webapp bug](https://mattermost.atlassian.net/browse/MM-50398) preventing such errors from showing if the slash command is sent in a reply box (in a RHS that is). To fix this we'd need changes on the other side.

Now, given the state of things I am starting to think we'd be better implementing our own mechanism for error surfacing, maybe by re-using the existing modals we have for "blocking errors". I created  [MM-50429](https://mattermost.atlassian.net/browse/MM-50429) to review all of this.